docker/login/mint-leaf.yml

```diff
@@ -0,0 +1,66 @@
+name: docker/login
+version: 1.0.0
+description: Log in to a Docker registry
+source_code_url: https://github.com/rwx-research/mint-leaves/tree/main/docker/login
+issue_tracker_url: https://github.com/rwx-research/mint-leaves/issues
+
+parameters:
+  username:
+    description: "The username to log in with."
+    required: true
+  password-env-name:
+    description: "The environment variable name that contains the Docker registry token or password."
+    default: DOCKER_PASSWORD
+  registry:
+    description: "The Docker registry to log in to."
+    default: docker.io
+
+tasks:
+  - key: produce-login-hooks
+    cache: false
```

Comment:
@tonywok can this be cached? Does `MINT_HOOKS_BEFORE_TASK` and `MINT_HOOKS_AFTER_TASK` work with cache hits?

Reply:
They do work with cache hits so I think that should be cacheable